pandas/_libs/tslib.pyx

```diff
@@ -1538,6 +1497,68 @@ cdef convert_to_tsobject(object ts, object tz, object unit,
     return obj
 
 
+cdef _TSObject convert_datetime_to_tsobject(datetime ts, object tz,
+                                            int32_t nanos=0):
+    """
+    Extract datetime and int64 from any of:
+        - python datetime object
```

Comment:
can u put a proper doc string

Reply:
This is adapted from the `convert_to_tsobject` docstring.

Reply:
take the opportunity to improve it